From: Boyuan Yang <byang@debian.org>
Date: Thu, 11 Jul 2024 00:08:48 -0400
Subject: Do not compile AArch64 SME kernels for now
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

... due to the current GCC 13 in Debian not recognizing the related
flags:

    cc1plus: error: invalid feature modifier ‘sme’ in ‘-march=armv9-a+sme’
    cc1plus: note: valid arguments are: fp simd crc lse fp16 rcpc rdma dotprod aes sha2 crypto sha3 sm4 fp16fml sve profile rng memtag sb ssbs predres sve2 sve2-sm4 sve2-aes sve2-sha3 sve2-bitperm tme i8mm f32mm f64mm bf16 flagm pauth ls64 mops cssc; did you mean ‘sm4’?
    make[3]: *** [CMakeFiles/yuv_sme.dir/build.make:79: CMakeFiles/yuv_sme.dir/source/rotate_sme.cc.o] Error 1

This bug shall be revised when new GCC is available in Debian.
---
 CMakeLists.txt | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f51166b..7b51f49 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -113,11 +113,7 @@ if(NOT MSVC)
     TARGET_COMPILE_OPTIONS(${ly_lib_name}_sve PRIVATE -march=armv9-a+sve2)
     LIST(APPEND ly_lib_parts $<TARGET_OBJECTS:${ly_lib_name}_sve>)
 
-    # Enable AArch64 SME kernels.
-    ADD_LIBRARY(${ly_lib_name}_sme OBJECT
-      ${ly_src_dir}/rotate_sme.cc)
-    TARGET_COMPILE_OPTIONS(${ly_lib_name}_sme PRIVATE -march=armv9-a+sme)
-    LIST(APPEND ly_lib_parts $<TARGET_OBJECTS:${ly_lib_name}_sme>)
+    # Debian-specific: Do not use AArch64 SME kernels for now.
   endif()
 endif()
 
